### PR TITLE
updates to go into 0.5.0 to fix bug #677

### PR DESCRIPTION
**Pull request by kylebechtel77**
fixed conditionals. tests now pass

### CODE REVIEW

**Comment by kylebechtel77** [comment]
Do not pull this in yet, just found an issue. working to resolve it now.

**Comment by kylebechtel77** [comment]
Issue is resolved. this can be pulled in now.

File: include/localization.js [hunk 1]
```diff
@@ -153,7 +157,14 @@ module.exports = function LocalizationModule(pb) {
         var tmp;
         var val = null;
         var loc = Localization.storage[this.language];
-        for (var category in loc) {
+
+        if(loc && loc[this.activeTheme] && loc[this.activeTheme][key])
+        {
+          return loc[this.activeTheme][key];
+        }
```
Comment (brianhyder) [inline include/localization.js]: Item 1

File: include/localization.js [hunk 2]
```diff
@@ -312,31 +323,27 @@ module.exports = function LocalizationModule(pb) {
...
+        pluginName = 'generic';
+
+      var localeObj = Localization.getLocalizationPackage(locale);
+
+      if (localeObj[pluginName] !== undefined) {
+        localeObj[pluginName][key] = value;
+      }
+      else {
+        localeObj[pluginName] = {};
+        localeObj[pluginName][key] = value;
```
Comment (brianhyder) [inline include/localization.js]: Item 2

**Comment by brianhyder** [comment]
Hey all, I wanted to pass along some feedback on this PR.  We won't be merging this in but I wanted to take a minute to explain why.  

1. There is a bug at "Item 1".  The code find the key and returns the value if there is a plugin override.  This breaks current functionality where parameterized values are not replaced when overrided by a plugin.
2. The data structure used to store plugin overrides has the potential to cause collisions with top level keys in the localization files.
3. There are API end-points that provide the localizations client side.  These would need to be updated too in order to provide backward compatibility as well as the ability to apply plugin overrides through some query string parameter.
4. There are already flaws in our current architecture with localization.  Specifically, the way we fail to distinguish between language by itself and when a country code is paired with a language code.  Fixing this will require a different data structure that splits out language, country, and themes.  There will also be additional complexity around multi-site and how that is handled.

Part of the work I am targeting for this quarter revolves around fixing these flaws and building language capabilities into the routes.  I really want to get the multi-site merged in before this work begins to avoid nasty merges and to create a base-line.  

I'd like to use this code as the starting point for that work.  The overall feel is that the fix for the bug is a part of a larger issue with the way things are structured.  

Let me know if y'all have any questions.

**Pull request closed without merging**